upgrade/util.go

```diff
@@ -47,6 +47,7 @@ type Context struct {
 	//	pulumi-aws
 	//
 	UpstreamProviderName string
+	UpstreamProviderOrg  string
```

Comment:
This was confusing, so let's document it.

```suggestion
	// The org component in the upstream provider's repo path.
	// 
	// This must be provided when upstream is hosted at a repo that does 
	// not match the Go Module path it is declared as. Otherwise it can 
	// be inferred.
	//
	// For example, if we have an upstream provider with import path:
	//
	//	github.com/terraform-providers/terraform-provider-my-provider
	//
	// which is hosted at:
	//
	//	github.com/my-org/terraform-provider-my-provider
	//
	// Then UpstreamProviderOrg should be `my-org`.
	UpstreamProviderOrg  string
```